pandas/core/frame.py

```diff
@@ -3533,12 +3533,23 @@ def _combine_match_columns(self, other, func, level=None, fill_value=None):
                                    axes=[left.columns, self.index])
         return self._constructor(new_data)
 
-    def _combine_const(self, other, func, raise_on_error=True):
+    def _combine_const(self, other, func,
+                       raise_on_error=True, fill_value=None):
         if self.empty:
             return self
 
-        new_data = self._data.eval(func=func, other=other,
-                                   raise_on_error=raise_on_error)
+        if fill_value is not None:
```

Comment:
Do you mean that `df2.add(2, fill_value=0)` should act like `df2.add(2).fillna(0)`, instead of `df2.fillna(0).add(2)`?